### DISCUSSION TITLE
"sourceType":"module" breaks linting if an arrow function is present

### DISCUSSION THREAD

**Discussion by robcca** [eslint/eslint#14705] [Q&A]
I'm editing Javascript in VS Code, with the Eslint extension, as well as Eslint installed in Node modules. I need to have the "sourceType":"module" property in my eslintrc.json file in order to allow "import" statements.

Unfortunately, the sourceType option causes any file with an arrow function to have a linting error on the first arrow function and no other linting in that file:

```
Parsing error: Unexpected token =
```

Commenting out the sourceType option breaks linting in a file in another way: import statements then have the following error, with no other linting being done in that file:

```
Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
```

How can I resolve this dilemma?

**Comment by nzakas** [comment]
Can you reproduce the problem on eslint.org/demo?

Without seeing the actual code and config, there’s not enough info here to figure out what could be happening. SourceType doesn’t control whether arrow functions can be parsed, that’s what ecmaVersion does.

**Comment by aladdin-add** [comment]
To use esm and ES6( e.g. arrow functions), you will need to config:
```js
parserOptions: {ecmaVersion: 6, sourceType: "module"}
```

**Reply by robcca** [reply]
> To use esm and ES6( e.g. arrow functions), you will need to config:
> 
> ```js
> parserOptions: {ecmaVersion: 6, sourceType: "module"}
> ```

With those options I get "Parsing error: Unexpected token =" on the first arrow function and no further linting in that file.

**Reply by aladdin-add** [reply]
can you repro on the [demo page](https://eslint.org/demo), or fill our lovely [issue template](https://github.com/eslint/eslint/issues/new?assignees=&labels=bug%2C+repro%3Aneeded&template=BUG_REPORT.md&title=), thanks!
